third_party/kiwi/kiwi/be_read.h

```diff
@@ -64,6 +64,8 @@ kiwi_be_read_options(kiwi_be_startup_t *su,
 			kiwi_var_set(&su->database, KIWI_VAR_UNDEF, value, value_size);
 		else if (name_size == 12 && !memcmp(name, "replication", 12))
 			kiwi_var_set(&su->replication, KIWI_VAR_UNDEF, value, value_size);
+		else if (name_size == 8 && !memcmp(name, "options", 8))
+			kiwi_parse_pgoptions_and_update_vars(vars, value);
```

Comment:
We do not pass value_size here, but value have a semantics of buffer, not a string. Coincidentally there may be \0 byte. But it's not necessary to be this way.

Reply:
Is it possible to have a string without `'\0'` at the end? I use `wordexp` function wich needed for `'\0'`.

Reply:
You can copy memory to palloc-ed area with +1 extra byte.

Reply:
What is the best way to use `palloc` with kiwi? If I add `#include "postgres.h"` I got an error: `fatal error: postgres.h: No such file or directory`

Reply:
Sorry, I'm a bit biased toward pg. Just use malloc().

Reply:
Done